source/base/NexusApp.cc

```diff
@@ -10,10 +10,9 @@
 
 #include "NexusApp.h"
 
+#include "BatchSession.h"
```

Comment:
You'd need to include here the `GeometryBase` header if `DetectorConstruction.h` only has a forward declaration, as suggested above.